src/contract.rs

```diff
@@ -114,14 +118,39 @@ pub fn query<S: Storage, A: Api, Q: Querier>(deps: &Extern<S, A, Q>, msg: QueryM
         // Base
         QueryMsg::Balance { address, .. } => query_balance(&deps, &address),
         // todo TokenInfo
-        QueryMsg::Transfers /* todo rename TransferHistory */ { address, n, .. } => query_transactions(&deps, &address, n),
+        QueryMsg::Transfers /* todo rename TransferHistory */ { address, n, .. } => query_transactions(&deps, address, n.clone()),
         // Native
         // todo ExchangeRate
         // Other - Test
         _ => unimplemented!(),
     }
 }
 
+
+pub fn query<S: Storage, A: Api, Q: Querier>(deps: &Extern<S, A, Q>, msg: QueryMsg) -> QueryResult {
+
+    match msg {
+        // Base
+        QueryMsg::Balance {..} => authenticated_queries(&deps, &msg),
+        // todo TokenInfo
+        QueryMsg::Transfers {..}/* todo rename TransferHistory */ => authenticated_queries(&deps, &msg),
+        // Native
+        // todo ExchangeRate
+        // Other - Test
+        QueryMsg::Swap {nonce, ..} => query_swap(&deps, nonce),
+        _ => unimplemented!(),
+    }
```

Comment:
Here I'd either write a method on `msg` named `needs_authentication`, or just do this: (which is less clean but is faster to write and run:
```rust
match &msg { // notice the `&`
    QueryMsg::Swap {nonce, ..} => query_swap(&deps, nonce),
    _ => authenticated_queries(deps, msg),
}
```